core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -451,6 +452,43 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
     }
   }
 
+  /**
+   * Helper method to update quota types counts and quotaTypesEnabled flag.
+   * @param quotaTypeKey The QuotaTypes constant (e.g., QuotaTypes.UserClientIdQuotaEnabled)
+   * @param increment True to increment count, false to decrement
+   */
+  private def updateQuotaTypes(quotaTypeKey: Int, increment: Boolean): Unit = {
+    if (quotaTypeKey == QuotaTypes.NoQuotas) {
+      return
+    }
+    val previousQuotaTypesEnabled = quotaTypesEnabled
+
+    // Update activeQuotaTypes counts
+    activeQuotaTypes.compute(quotaTypeKey, (_, count) =>
+      if (increment) Option(count).getOrElse(0) + 1
+      else if (Option(count).exists(_ > 1)) count - 1
+      else 0
+    )
+
+    // Update quotaTypesEnabled based on activeQuotaTypes counts
```

Comment:
maybe would be helpful if this comment explained the bitwise operation, potentially with an example. 

e.g. quotaTypesEnabled is the bitwise representation of what quota types are enabled. For instance, if quotaTypesEnabled == 6, then we know we have UserClientIdQuotaEnabled and UserQuotaEnabled since their bit representations are 4 and 2. (4 | 2 = 6)